internal/daemon/worker/controller_connection.go

```diff
@@ -434,13 +434,13 @@ func monitorUpstreamConnectionState(ctx context.Context, cc *grpc.ClientConn, co
 	}
 
 	for cc.WaitForStateChange(ctx, state) {
-		newState := cc.GetState()
+		state := cc.GetState()
```

Comment:
I'm confused, how does this fix the issue? This code looks functionally identical to the old code. Is `state` supposed to overwrite the variable in the outer state? If so, I think you want
```suggestion
		state = cc.GetState()
```

Reply:
Yes my bad. I pushed an older code change. 

The new update should be `state = cc.GetState()` now